nixos/tests/installer.nix

```diff
@@ -68,160 +67,167 @@ let
                   , testCloneConfig
                   }:
     let
+
+      # FIXME taken from tests/boot.nix - should probably be moved to a common location
+      pythonDict = params: "\n    {\n        ${concatStringsSep ",\n        " (mapAttrsToList (name: param: "\"${name}\": \"${param}\"") params)},\n    }\n";
```

Comment:
We should get rid of this abomination. Just pass in a JSON file instead.

Reply:
As in using `json.loads` and `builtins.toJSON` ?

Reply:
yes

Reply:
Currently we get the following though
```py
--- /nix/store/3ybn9vy8l21prfh3c9w3143hdappa1j7-nixos-test-driver-boot-netboot-bios/test-script 2019-11-11 18:36:47.980602 +0000
+++ /nix/store/3ybn9vy8l21prfh3c9w3143hdappa1j7-nixos-test-driver-boot-netboot-bios/test-script 2019-11-11 18:36:48.133975 +0000
@@ -1,6 +1,14 @@
 import json
-machine = create_machine(json.reads({"netBackendArgs":"tftp=/nix/store/ysh6gx7n6za7pcm4388hv5pvlzpzwwsp-ipxeBootDir,bootfile=netboot.ipxe","qemuFlags":"-boot order=n -m 2000"}))
+
+machine = create_machine(
+    json.reads(
+        {
+            "netBackendArgs": "tftp=/nix/store/ysh6gx7n6za7pcm4388hv5pvlzpzwwsp-ipxeBootDir,bootfile=netboot.ipxe",
+            "qemuFlags": "-boot order=n -m 2000",
+        }
+    )
+)
 machine.start()
 machine.wait_for_unit("multi-user.target")
 machine.shutdown()

would reformat /nix/store/3ybn9vy8l21prfh3c9w3143hdappa1j7-nixos-test-driver-boot-netboot-bios/test-script
Oh no! 💥 💔 💥
1 file would be reformatted.
```
so probably best to actually use a file. In the future, when we use structured attributes, we would simply reads from `.attrs.json`.

Reply:
suppose we could also pass `indent=4`

Reply:
yeah just stumbled on that too.
gonna pick this up tomorrow

Reply:
let's see what happens with https://github.com/NixOS/nixpkgs/pull/73241

Reply:
@FRidh just pushed a change. what do you think?

doesn't the change proposed #73241 run into the same issue with the linter ?

Reply:
@FRidh  can this be considered as resolved?